src/IBM.Cloud.SDK.Core/Util/CredentialUtils.cs

```diff
@@ -83,27 +83,42 @@ private static VcapCredential GetVcapCredentialsObject(Dictionary<string, List<V
             {
                 return null;
             }
-
-            foreach (KeyValuePair<string, List<VcapCredential>> kvp in vcapServices)
-            {
-                if (kvp.Key.StartsWith(serviceName))
-                {
-                    List<VcapCredential> credentials = kvp.Value;
-                    if (!string.IsNullOrEmpty(plan))
-                    {
-                        foreach (VcapCredential credential in credentials)
-                        {
-                            if (credential.Plan.ToLower() == plan.ToLower())
-                            {
-                                return credential;
-                            }
-                        }
-                    }
-                    else
-                    {
-                        return credentials[0];
-                    }
-                }
+            // search the inner entries of the vcap for the matching serviceName
+            foreach (KeyValuePair<string, List<VcapCredential>> kvp in vcapServices)
+            {
+                List<VcapCredential> item = kvp.Value;
+                if (item != null && item.Count > 0)
+                {
+                  foreach (VcapCredential credential in item)
+                  {
+                    if (credential.Name == serviceName)
+                    {
+                      return credential;
+                    }
+                  }
+                } 
+            }
+            // Second, try to find a service list with the specified key.
+            if (vcapServices.TryGetValue(serviceName, out List<VcapCredential> credentials))
+            {
+                if (credentials == null || credentials.Count == 0)
+                {
+                    return null;
+                }
+                if (!string.IsNullOrEmpty(plan))
+                {
+                    foreach (VcapCredential credential in credentials)
+                    {
+                        if (credential.Plan.ToLower() == plan.ToLower())
```

Comment:
@padamstx @mediumTaj - I was skeptical about leaving this here. I want to say we don't check the plan in the other languages, but I may be wrong. Is this still being used?

Reply:
heh I think I copied this from Java. Looks like it was removed.

https://github.com/IBM/java-sdk-core/commit/f40c95229d3c86e7f8f76ba16cf88fb146500936#diff-5500372b768c88ebe4772f4eb00299afL148

I was trying to stay as close to Java as possible so if we don't use it there I think it's OK to remove in .NET as well

Reply:
Great, just wanted to double check with you. I'll go ahead and remove that check then